File: TGraph/random_walk.cpp
#include <TApplication.h>
#include <TCanvas.h>
#include <TGraph.h>
#include <TGraph2D.h>
#include <TRandom3.h>
#include <vector>

using namespace std;

void random_walk2d(const size_t N = 10000)
{

  TRandom3 rng{};
  auto xy_scatter = new TGraph();
  xy_scatter->SetTitle(";x;y");
  double x = 0, y = 0;
  for (size_t i = 0; i < N; ++i) {
    xy_scatter->SetPoint(i, x, y);
    x += (rng.Uniform() - 0.5) / 2.0;
    y += (rng.Uniform() - 0.5) / 2.0;
  }

  auto c = new TCanvas("2d");
  xy_scatter->Draw();
  c->Draw();
}

void random_walk3d(const size_t N = 10000)
{

  TRandom3 rng{};

  vector<double> x(N), y(N), z(N);
  x[0] = y[0] = z[0] = 0.0;
  for (size_t i = 0; i < N; ++i) {
    x[i] = x[i - 1] + (rng.Uniform() - 0.5) / 2.0;
    y[i] = y[i - 1] + (rng.Uniform() - 0.5) / 2.0;
    z[i] = z[i - 1] + (rng.Uniform() - 0.5) / 2.0;
  }

  auto xyz_scatter = new TGraph2D(N, x.data(), y.data(), z.data());
  xyz_scatter->SetTitle(";x;y;z");

  auto c = new TCanvas("3d");
  xyz_scatter->Draw("LINE");
  c->Draw();
}

int main(int agrc, char* argv[])
{
  TApplication app("Random walk", nullptr, nullptr);

  random_walk2d();
  random_walk3d();
  app.Run();

  return 0;
}

File: TGraph/random_walk.py
from ROOT import TRandom3, TCanvas, TGraph, TGraph2D, gROOT

def random_walk2d(N = 1000):
    rng = TRandom3()
    xy_scatter = TGraph()
    xy_scatter.SetTitle(";x;y")
    x = 0; y = 0
    for i in range(0, N):
        xy_scatter.SetPoint(i, x, y)
        x += (rng.Uniform() - 0.5) / 2.0;
        y += (rng.Uniform() - 0.5) / 2.0;

    c = TCanvas("2d")
    xy_scatter.Draw()
    c.Draw()
    input('Press Enter to close')

def random_walk3d(N = 1000):
    rng = TRandom3()
    xyz_scatter = TGraph2D()
    xyz_scatter.SetTitle(";x;y;z")
    x = 0.0; y = 0.0; z = 0.0
    for i in range(0, N):
        xyz_scatter.SetPoint(i, x, y, z)
        x += (rng.Uniform() - 0.5) / 2.0;
        y += (rng.Uniform() - 0.5) / 2.0;
        z += (rng.Uniform() - 0.5) / 2.0;

    c = TCanvas("3d")
    xyz_scatter.Draw("LINE")
    c.Draw()
    input('Press Enter to close')


def main():
    gROOT.SetBatch(False)
    random_walk2d()
    random_walk3d()

if __name__ == "__main__":
    main()
